lib/api/src/externals/function.rs

```diff
@@ -890,8 +910,10 @@ impl<T: VMDynamicFunction> VMDynamicFunctionCall<T> for VMDynamicFunctionContext
         let result = panic::catch_unwind(AssertUnwindSafe(|| {
             let func_ty = self.ctx.function_type();
             let mut args = Vec::with_capacity(func_ty.params().len());
+            // TODO: we need a store here, where do we get it?
+            let hack = 3;
```

Comment:
?

Reply:
This is the `Value` change mentioned above:

>  Passing a store via &dyn Any to some methods and the garbage placeholder values we pass when we don't have a store.

Currently because of the way that Value works we need a `Store` when dealing with `FuncRef`. Additionally because `Value` doesn't know about `Store` due to the shape of our dependency tree, we have to pass it secretly using traits and `dyn Any`. Here we don't have access to a `Store` (but probably need one). This was a hack to get it to compile but I believe the fact that it hasn't failed any of our tests yet is a symptom of a lack of tests, specifically tests calling functions that pass and return `FuncRef`s as values.